subprojects/dependency-management/src/main/java/org/gradle/api/internal/artifacts/transform/DefaultTransformerInvoker.java

```diff
@@ -272,29 +318,22 @@ public void store(Consumer<BuildCacheKey> storer) {
 
         @Override
         public void persistResult(ImmutableSortedMap<String, CurrentFileCollectionFingerprint> finalOutputs, boolean successful, OriginMetadata originMetadata) {
-            getResult().ifSuccessful(result -> {
-                historyRepository.persist(cacheKey.getPersistentCacheKey(), result, finalOutputs.values().iterator().next());
-            });
+            historyRepository.persist(cacheKey.getPersistentCacheKey(),
+                originMetadata,
+                // TODO: only use implementation hash
+                ImplementationSnapshot.of(transformer.getImplementationClass().getName(), transformer.getSecondaryInputHash()),
+                finalOutputs,
+                successful
+            );
```

Comment:
If we don't, the history will still claim the last run was successful, right?

I guess my thinking was wrong: for tasks, we store the history even if the task failed since we don't want to have false positives for overlapping output detection. We don't need to do this for transformations, right?